### ISSUE TITLE
react/react-compiler reports PruneHoistedContexts internal errors for recursive closures

### ISSUE THREAD

**Issue by jonathanong** [oxc-project/oxc#24374]
## Description

Oxlint native `react/react-compiler` reports an internal compiler error instead of a normal diagnostic for hooks containing recursive closures:

```
Unexpected error: Todo: [PruneHoistedContexts] Rewrite hoisted function references
```

This occurs in two production hooks under Oxlint 1.43.0. One recursively schedules a retry from inside a `setTimeout` promise chain; the other defines several hook-local async handlers and a reset helper.

## Expected behavior

The rule should either compile these hooks or emit an actionable user-code diagnostic, not an internal `Todo` error.

## Environment

- Oxlint: 1.43.0
- Rule: native `react/react-compiler`
- Type-aware mode enabled

**Comment by jonathanong** [comment]
sorry, AI. will try to create a better issue later if it's still an issue
